src/us/kbase/narrativejobservice/sdkjobs/SDKMethodRunner.java

```diff
@@ -681,13 +684,41 @@ public static JobState checkJob(String jobId, AuthToken authPart,
 
 	public static CheckJobsResults checkJobs(CheckJobsParams params, AuthToken auth,
 	        Map<String, String> config) throws Exception {
-	    CheckJobsResults ret = new CheckJobsResults().withJobStates(new LinkedHashMap<String, JobState>());
-	    for (String jobId : params.getJobIds())
-	        ret.getJobStates().put(jobId, checkJob(jobId, auth, config));
-	    if (params.getWithJobParams() != null && params.getWithJobParams() == 1L) {
-	        ret.withJobParams(new LinkedHashMap<String, RunJobParams>());
-	        for (String jobId : params.getJobIds())
-	            ret.getJobParams().put(jobId, getJobInputParams(jobId, auth, config, null));
+	    if (params.getJobIds() == null) {
+	        throw new IllegalStateException("Input parameters should include 'job_ids' property");
+	    }
+	    boolean withJobParams = params.getWithJobParams() != null && 
+	            params.getWithJobParams() == 1L;
+	    Map<String, RunJobParams> jobParams = new LinkedHashMap<String, RunJobParams>();
+	    Map<String, JsonRpcError> checkError = new LinkedHashMap<String, JsonRpcError>();
+	    CheckJobsResults ret = new CheckJobsResults().withJobStates(
+	            new LinkedHashMap<String, JobState>()).withCheckError(checkError);
+	    for (String jobId : params.getJobIds()) {
+	        try {
+	            ret.getJobStates().put(jobId, checkJob(jobId, auth, config));
+	            if (withJobParams) {
+	                jobParams.put(jobId, getJobInputParams(jobId, auth, config, null));
+	            }
+	        } catch (Exception ex) {
```

Comment:
This seems like a backwards incompatible change - I assume that this method used to throw an exception if one check failed? Is not throwing an exception going to break code that assumes an exception will be thrown if one of the job check fails?